### DISCUSSION TITLE
[Backend] Kotlin + Spring-boot + gRPC + PostgreSQL

### DISCUSSION THREAD

**Discussion by alejandrolorefice** [realworld-apps/realworld#1317] [WIP Implementations]
Hello! :wave:

I've started working on a Kotlin gRPC implementation of the real-world example. 

I get the feeling that _gRPC_ isn't as widely adopted as I think it deserves and specific documentation or examples about it are rather scarce compared to _REST_ or _GraphQL_.

Also, most of the information I come across online is using _Go_ or _.NET_ while _Kotlin_ is relatively rare. For example, a quick search at Google throws 400k results for "Kotlin gRPC" while I get nearly 7M and 5M if I replace _Kotlin_ with _Go_ or _.NET_, respectively.

So, hopefully, this helps thicken the _gRPC_ :handshake: _Kotlin_ information available and helps other people wanting to give them a try.

Cheers!

**Comment by geromegrignon** [comment]
Hello Alejandro,

Thanks for your contribution.
Promoting a project is clearly our goal.
However right now we won't be able to list it in the official RealWorld listing as based on our current specs.
Part of our contract is to ensure frontend projects can work with backend ones.

I'm keeping the idea in mind though. We already had a request for GraphQL too and due to the popularity of both of them, extending the current RealWorld project is doable. It would just need some time to do it.

**Reply by alejandrolorefice** [reply]
Hey Gerome! Yeah, I wasn't assuming full gRPC support. I plan to make it REST compatible as [other gRPC implementations did](https://github.com/raahii/golang-grpc-realworld-example). Although having also the pure gRPC interface would be very interesting indeed :thinking:
